.ci/bumpGoReleaseVersion.groovy

```diff
@@ -0,0 +1,149 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+@Library('apm@master') _
+
+pipeline {
+  agent { label 'linux && immutable' }
+  environment {
+    REPO = 'apm-pipeline-library'
+    ORG_NAME = 'elastic'
+    HOME = "${env.WORKSPACE}"
+    NOTIFY_TO = credentials('notify-to')
+    PIPELINE_LOG_LEVEL = 'INFO'
+  }
+  options {
+    timeout(time: 1, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '20', artifactNumToKeepStr: '20'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+    rateLimitBuilds(throttle: [count: 60, durationName: 'hour', userBoost: true])
+    quietPeriod(10)
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        git(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken', url: "https://github.com/${ORG_NAME}/${REPO}.git")
+      }
+    }
+    stage('Fetch latest go release version') {
+      steps {
+        setEnvVar('GO_RELEASE_VERSION', goVersion(action: 'latest', unstable: false))
```

Comment:
This is not filtering by any releases but the latest.

If it's required every subscribed project can define their `go<major>.<minor>`  in case they only care about patches. 

In other words, if `1.16` is the current go line but `1.17` is released then this current implementation will pick `1.17`.  the `blob` argument in the `goVersion` step provides the feature to filter.

If the above is needed, then I'd say this particular stage ('Fetch Latest go Release version') should executed within the context of the project/branch, and `.ci/.bump-go-release-version.yml` might contain the `go` version line supported for each branch. Therefore, this file will need to keep in sync with the go<major>.<minor>